vcd/resource_vcd_catalog.go

```diff
@@ -182,6 +212,13 @@ func genericResourceVcdCatalogRead(d *schema.ResourceData, meta interface{}) err
 		return fmt.Errorf("error retrieving catalog %s : %s", d.Id(), err)
 	}
 
+	// Catalog record is retrieved to get the number of vApp templates, medias and owner name
+	adminCatalogRecord, err := adminOrg.FindAdminCatalogRecords(adminCatalog.AdminCatalog.Name) // Is this returning one even though there are catalogs with similar names?
```

Comment:
why in datasource `QueryCatalogList` and here `FindAdminCatalogRecords` is used?

Reply:
In datasource we do not retrieve the admin view from a catalog. Also there is no function for finding catalog records as it exists for admin catalog.
In the resource, since the only way to create a catalog is the admin view, we can use that function as it is available.

Reply:
We had investigated and agreed that there is a need a little polish with methods in govcd part.

Reply:
@vbauzysvmware already open a PR for this in go-vcd. Please check https://github.com/vmware/go-vcloud-director/pull/450
With that, we will be able to get specific catalogRecord by catalog name.

Also, to make the code more consistent, both System Administrator and Org Administrator can query the same way AdminCatalog, so what I've done is to change the `Catalog` struct being used in the datasource to `AdminCatalog` struct. This way code is more consistent.

Reply:
Thank you